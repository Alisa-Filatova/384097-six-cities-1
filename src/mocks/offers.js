export default [
  {
    id: 123,
    title: `Beautiful & luxurious apartment at great location`,
    img: `img/apartment-01.jpg`,
    isPremium: true,
    price: 120,
    stars: 4,
    type: `Apartment`,
    isInBookmarks: false,
  },
  {
    id: 234,
    title: `Wood and stone place`,
    img: `img/apartment-02.jpg`,
    isPremium: false,
    price: 80,
    stars: 2,
    type: `Private room`,
    isInBookmarks: true,
  },
  {
    id: 456,
    title: `Canal View Prinsengracht`,
    img: `img/apartment-03.jpg`,
    isPremium: false,
    price: 180,
    stars: 3,
    type: `Apartment`,
    isInBookmarks: false,
  },
  {
    id: 567,
    title: `Nice, cozy, warm big bed apartment`,
    img: `img/apartment-01.jpg`,
    isPremium: false,
    price: 100,
    stars: 4,
    type: `Private room`,
    isInBookmarks: true,
  },
];
