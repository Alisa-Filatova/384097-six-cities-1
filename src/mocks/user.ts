export const userMock = {
  avatarUrl: `/static/avatar/4.jpg`,
  email: `alicefill@y888.ru`,
  id: 1,
  isPro: false,
  name: `alicefill`,
};
