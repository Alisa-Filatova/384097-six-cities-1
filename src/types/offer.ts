export interface City {
  name: string,
  location: {
    latitude: number,
    longitude: number,
    zoom: number,
  }
}

export interface Host {
  id: number;
  isPro: boolean;
  name: string;
  avatarUrl: string;
}

export interface OfferLocation {
  latitude: number;
  longitude: number;
  zoom: number;
}

export interface Offer {
  id: number;
  city: City;
  previewImage: string;
  images: [string];
  title: string;
  isFavorite: boolean;
  isPremium: boolean;
  rating: number;
  type: string;
  bedrooms: number;
  maxAdults: number;
  price: number;
  goods: [string];
  host: Host,
  description: string;
  location: OfferLocation;
}

export interface CityOffersGroup {
  [key: string]: Offer[];
}
