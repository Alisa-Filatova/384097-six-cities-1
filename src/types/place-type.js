export const PlaceType = {
  house: `House`,
  hotel: `Hotel`,
  apartment: `Apartment`,
  room: `Private Room`,
};
