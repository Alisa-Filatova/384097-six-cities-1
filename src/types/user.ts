export interface User {
  id: number;
  email: string;
  name: string;
  avatarUrl: string;
  isPro: boolean;
}

export interface Review {
  id: number;
  user: User;
  rating: number;
  comment: string;
  date: string;
}

export interface Comment {
  rating: number;
  comment: string;
}
