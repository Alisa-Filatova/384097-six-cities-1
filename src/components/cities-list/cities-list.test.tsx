import * as React from 'react';
import * as renderer from 'react-test-renderer';
import CitiesList from './cities-list';
import {cityMock} from '../../mocks/city';
import {City} from '../../types/offer';

describe('CitiesList', () => {
  it('renders correctly', () => {
    const list = renderer.create(
        <CitiesList
          cities={[cityMock] as City[]}
          currentCity={cityMock}
          onCityClick={jest.fn()}
        />
    ).toJSON();
    expect(list).toMatchSnapshot();
  });
});
