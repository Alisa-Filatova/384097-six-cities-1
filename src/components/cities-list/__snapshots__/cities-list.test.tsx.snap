// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesList renders correctly 1`] = `
<div
  className="cities tabs"
>
  <section
    className="locations container"
  >
    <div
      className="locations__list tabs__list"
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item tabs__item--active"
          href="#"
          onClick={[Function]}
          style={
            Object {
              "pointerEvents": "auto",
            }
          }
        >
          <span>
            Amsterdam
          </span>
        </a>
      </div>
    </div>
  </section>
</div>
`;
